### ISSUE TITLE
arangodump gone wild

### ISSUE THREAD

**Issue by pkoppstein** [arangodb/arangodb#4184]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.3

Mode:
- [x] Single-Server

Storage-Engine:
- [x] mmfiles

On this operating system:
- [x] MacOS, version: High Sierra

## These are the steps to reproduce:

* arangodump --output-directory dump_all

The following problems occur:
1) A large collection of extremely simple documents causes the following failure:

2017-12-30T05:06:49Z [22848] ERROR got invalid response from server: Request timeout reached
Processed 14 collection(s), wrote 107320873 byte(s) into datafiles, sent 18 batch(es)

2) I thought that was the end of it, but the CPU started overheating, and the arangod process was consuming humungous amounts of memory, so I gracefully terminated it, and saw these messages:

2017-12-21T05:27:52Z [22265] INFO ArangoDB (version 3.3.0 [darwin]) is ready for business. Have fun!
2017-12-30T07:00:12Z [22265] INFO control-c received, beginning shut down sequence
2017-12-30T07:00:15Z [22265] WARNING caught exception while filling indexes: shutdown in progress
2017-12-30T07:00:30Z [22265] ERROR caught exception during dump of collection 'connections': invalid document type

**Note that the WARNING and ERROR messages were only emitted AFTER the shut down message!**

3) And I thought that was the end of it again, but the "connections" collection is evidently corrupted - the web interface shows "loading" when I click on it.  Previously, this collection loaded almost instantaneously.

Here is the output of `arangodump --version`
3.3.0

architecture: 64bit
arm: false
asan: false
asm-crc32: false
assertions: false
boost-version: 1.62.0
build-date: 2017-12-20 15:01:58
compiler: clang [4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
cplusplus: 201103
debug: false
endianness: little
failure-tests: false
fd-client-event-handler: poll
fd-setsize: 1024
full-version-string: ArangoDB 3.3.0 [darwin] 64bit, using jemalloc, VPack 0.1.30, RocksDB 5.6.0, ICU 58.1, V8 5.7.492.77, OpenSSL 1.0.2n  7 Dec 2017
icu-version: 58.1
jemalloc: true
maintainer-mode: false
openssl-version: OpenSSL 1.0.2n  7 Dec 2017
platform: darwin
reactor-type: kqueue
rocksdb-version: 5.6.0
server-version: 3.3.0
sizeof int: 4
sizeof void*: 8
sse42: false
unaligned-access: true
v8-version: 5.7.492.77
vpack-version: 0.1.30
zlib-version: 1.2.11

**Comment by pkoppstein** [comment]
Here is some supplementary information regarding the corrupted collection, named "connections".

In brief, `db.connections.properties()` succeeded, but just about everything else failed (quietly). However, I was able to drop() the database.

```
127.0.0.1:8529@_system> db.connections.properties()
{ 
  "doCompact" : true, 
  "journalSize" : 33554432, 
  "isSystem" : false, 
  "isVolatile" : false, 
  "waitForSync" : false, 
  "keyOptions" : { 
    "type" : "traditional", 
    "allowUserKeys" : true, 
    "lastValue" : 0 
  }, 
  "indexBuckets" : 8 
}

127.0.0.1:8529@_system> db.connections.figures()
db.connections.figures()
# hangs

127.0.0.1:8529@_system> db.connections.count()
db.connections.count()
# hangs

127.0.0.1:8529@_system> db.connections.revision()
db.connections.revision()
 # hangs

```
I also tried restarting arangod with
      `--database.ignore-datafile-errors true --wal.ignore-logfile-errors true`
but that didn't make any difference.

**Comment by dothebart** [comment]
Hi,
are you using the same dump to restore to more than one database?

**Comment by pkoppstein** [comment]
I never did any restoration.  That is, I ran arangodump and then noticed that the collection which caused arangodump to fail was corrupt.  

Unfortunately, the cause-and-effect chain is unclear to me.  I was hoping that the details I gave might help in that regard.  It's certainly possible that the failure of arangodump was a symptom of the corruption rather than the cause, but I would have hoped that if arangodump was presented with a corrupt collection, it would say so and give some diagnostics.

Thanks!
